fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1694,6 +1694,29 @@ public Set<String> getVisiblePartitionNames() {
                 .collect(Collectors.toSet());
     }
 
+    public String getLatestHasDataPartitionValue() throws DdlException {
+        if (partitionInfo.getType() != PartitionType.RANGE) {
+            throw new DdlException("Currently only range partitions are supported");
+        }
+
+        List<Partition> hasDataPartitions =
+                idToPartition.values().stream().filter(Partition::hasData).toList();
+        if (hasDataPartitions.isEmpty()) {
+            return null;
+        }
+        RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+        List<Range<PartitionKey>> rangePartitions = new ArrayList<>();
+        for (Partition partition : hasDataPartitions) {
+            Range<PartitionKey> range = rangePartitionInfo.getRange(partition.getId());
+            rangePartitions.add(range);
+        }
+        List<Range<PartitionKey>> sortedRange =
```

Comment:
`PartitionInfo#getSortedPartitions` has a similar functionality, it's better to reuse it

Reply:
good idea.